### PR TITLE
Fix #788 - added option to view receipt when downloaded

### LINKED ISSUE #788: Feat: add an option to view downloaded receipt in ReceiptActivity
**Summary:**
When the receipt is downloaded in the `ReceiptActivity` only the successful download message is displayed I think I would be better if we could also give the user the option to view the downloaded receipt

### PR DESCRIPTION
**Pull request by dubeyaayush07**
Fix #788

Screenrecorder-2020-03-03-22-56

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by dubeyaayush07** [comment]
@shiv07tiwari after the recent merge there were too many merge conflicts in #789 and as only a little bit of code was involved I decided to close #789 and create a new PR

**Comment by dubeyaayush07** [comment]
@luckyman20 please review

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java [hunk 1]
```diff
@@ -195,14 +198,35 @@ public void writeReceiptToPDF(ResponseBody responseBody, String filename) {
         if (!mifosDirectory.exists()) {
             mifosDirectory.mkdirs();
         }
-        File documentFile = new File(mifosDirectory.getPath(), filename);
+        final File documentFile = new File(mifosDirectory.getPath(), filename);
         if (!FileUtils.writeInputStreamDataToFile(responseBody.byteStream(), documentFile)) {
             showToast(Constants.ERROR_DOWNLOADING_RECEIPT);
         } else {
-            showSnackbar(Constants.RECEIPT_DOWNLOADED_SUCCESSFULLY);
+            Snackbar.make(findViewById(android.R.id.content),
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java:205]: Please the utils lib `Toaster` that we have in the project.

Comment (dubeyaayush07) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java:205]: @luckyman20 I initially tried using the Toaster class but it did not have the method for creating snackbar with an action so I had to use the Snackbar class directly.

Comment (luckyman20) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java:205]: Add that in the `Toaster` class and then use it. Don't forget to mention it in commit description.

**Comment by dubeyaayush07** [comment]
@luckyman20 please review

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java [hunk 2]
```diff
@@ -195,14 +198,34 @@ public void writeReceiptToPDF(ResponseBody responseBody, String filename) {
...
+    private void openFile(Context context, File file) {
+        final Uri data = FileProvider.getUriForFile(context,
+                "org.mifos.mobilewallet.mifospay.provider", file);
+        context.grantUriPermission(context.getPackageName(),
+                data, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        Intent intent = new Intent(Intent.ACTION_VIEW)
+                .setDataAndType(data, "application/pdf")
+                .addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+        intent = Intent.createChooser(intent, "View Receipt");
```
Comment (jawidMuhammadi) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java:225]: `View Receipt` is a UI related text and it needs to be translated, so please add it in `string.xml` file

Comment (luckyman20) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java:225]: @dubeyaayush07 Please look into this.

Comment (dubeyaayush07) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java]: @luckyman20 Done

**Comment by dubeyaayush07** [comment]
@luckyman20 I have made the changes you asked for and the Toaster now has a method for accepting snackbar actions

**Comment by shiv07tiwari** [comment]
Good work @dubeyaayush07

**Pull request merged**